File: Roman to Integer/Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        int sum = 0;
        for(unsigned int i = 0; i < s.length(); ++i){
            char digit = s[i];
            switch(digit){
                case 'M':
                    sum += 1000;
                    break;
                case 'D':
                    sum += 500;
                    break;
                case 'C':
                    if(i < s.length() - 1){     // Make sure that 'C' is not LSB
                        if((s[i + 1] == 'D') || (s[i + 1] == 'M')){
                            sum -= 100;
                        }
                        else{
                            sum += 100;
                        }
                    }
                    else{
                        sum += 100;
                    }
                    break;
                case 'L':
                    sum += 50;
                    break;
                case 'X':
                    if(i < s.length() - 1){
                        if((s[i + 1] == 'L') || (s[i + 1] == 'C')){
                            sum -= 10;
                        }
                        else{
                            sum += 10;
                        }
                    }
                    else{
                        sum += 10;
                    }
                    break;
                case 'V':
                    sum += 5;
                    break;
                case 'I':
                    if(i < s.length() - 1){
                        if((s[i + 1] == 'V') || (s[i + 1] == 'X')){
                            sum--;
                        }
                        else{
                            sum++;
                        }
                    }
                    else{
                        sum++;
                    }
                    break;
                default:
                
                    break;
            }
        }
        return sum;
    }
};

File: Roman to Integer/Roman to Integer.java
public class Solution {
    public int romanToInt(String s) {
        int res = 0;
        for(int i = 0; i < s.length(); ++i){
            switch(s.charAt(i)){
                case 'I':
                    if(i == s.length() - 1){
                        ++res;
                    }
                    else if(s.charAt(i + 1) == 'V' || s.charAt(i + 1) == 'X'){
                        --res;
                    }
                    else{
                        ++res;
                    }
                    break;
                case 'V':
                    res += 5;
                    break;
                case 'X':
                    if(i == s.length() - 1){
                        res += 10;
                    }
                    else if(s.charAt(i + 1) == 'L' || s.charAt(i + 1) == 'C'){
                        res -= 10;
                    }
                    else{
                        res += 10;
                    }
                    break;
                case 'L':
                    res += 50;
                    break;
                case 'C':
                    if(i == s.length() - 1){
                        res += 100;
                    }
                    else if(s.charAt(i + 1) == 'D' || s.charAt(i + 1) == 'M'){
                        res -= 100;
                    }
                    else{
                        res += 100;
                    }
                    break;
                case 'D':
                    res += 500;
                    break;
                case 'M':
                    res += 1000;
                    break;
            }
        }
        
        return res;
    }
}
